python/cudf/cudf/core/frame.py

```diff
@@ -337,23 +332,30 @@ def _slice(self, arg):
         if stop < 0:
             stop = stop + num_rows
 
-        if start > stop:
+        if start > stop and (stride is None or stride == 1):
             return self._empty_like(keep_index)
         else:
             start = len(self) if start > num_rows else start
             stop = len(self) if stop > num_rows else stop
 
-            result = self._from_table(
-                libcudf.copying.table_slice(self, [start, stop], keep_index)[0]
-            )
+            if stride is not None and stride != 1:
+                return self._gather(
+                    cupy.arange(start, stop=stop, step=stride, dtype=np.int64)
```

Comment:
Can we use `int32` for this? It would be more optimal for the gather map.

Reply:
Forgot about `size_type`